crates/recorder/src/benchmark.rs

```diff
@@ -94,46 +98,34 @@ impl<'a> Module<'a> {
 }
 
 /// An instantiated module from our `BenchApi`.
-pub struct Instance<'a> {
-    engine: Engine<'a>,
+pub struct Instance<'a, 'b> {
+    engine: Engine<'a, 'b>,
 }
 
-impl<'a> Instance<'a> {
+impl<'a, 'b> Instance<'a, 'b> {
     /// Execute this instance's code, returning the engine it was in.
     pub fn execute(self, measure: Box<dyn Measure>) -> Measurement {
         static_measure::setup(measure);
-        let result = unsafe { (self.engine.bench_api.engine_execute_module)(self.engine.engine) };
+        let result = unsafe { (self.engine.bench_api.wasm_bench_execute)(self.engine.engine) };
         let execution = static_measure::result();
         assert_eq!(result, 0);
         execution
     }
 }
 
 /// Measure various phases of a Wasm module's lifetime. This relies on `engine_lib_path`, the path
-/// to a shared library that implements `engine_*` functions that allow for measurement of each
+/// to a shared library that implements `wasm_bench_*` functions that allow for measurement of each
 /// phase.
 pub fn benchmark(
-    wasm_bytes: impl AsRef<[u8]>,
-    engine_lib_path: impl AsRef<OsStr>,
+    bench_api: &mut BenchApi,
+    wasm_bytes: &[u8],
     measure_type: MeasureType,
 ) -> Result<BenchmarkMeasurements> {
-    let wasm_bytes = wasm_bytes.as_ref();
-    info!(
-        "Attempting to run benchmark of {} bytes with engine at {:?}",
-        wasm_bytes.len(),
-        engine_lib_path.as_ref()
-    );
```

Comment:
Can we retain some level of logging in here? Like maybe an info in `BenchApi::new` telling us the path of the library we just loaded and maybe something here indicating that we are starting a new benchmark run (e.g. number of bytes, measure type). I find logging extremely helpful when I'm too lazy to debug.